lib/instrumentation/transaction.js

```diff
@@ -218,3 +222,71 @@ Transaction.prototype.end = function (result, endTime) {
   this._agent._instrumentation.addEndedTransaction(this)
   this._agent.logger.debug('ended transaction %o', { trans: this.id, parent: this.parentId, trace: this.traceId, type: this.type, result: this.result, name: this.name })
 }
+
+Transaction.prototype._captureBreakdown = function (span) {
+  const metrics = this._agent._metrics
+
+  // Record span data
+  captureBreakdown(this, {
+    transaction: transactionBreakdownDetails(this),
+    span: spanBreakdownDetails(span)
+  }, span._timer.selfTime)
+
+  // Record transaction data
+  if (span instanceof Transaction) {
+    const { duration } = this._timer
+    const labels = flattenBreakdown({
+      transaction: transactionBreakdownDetails(this)
+    })
+
+    metrics.getOrCreateCounter('transaction.duration.count', labels).inc()
+    metrics.getOrCreateCounter('transaction.duration.sum.us', labels).inc(duration)
+    metrics.getOrCreateCounter('transaction.breakdown.count', labels).inc(this.sampled ? 1 : 0)
+
+    this._breakdownTimings.forEach(({ labels, value }) => {
+      labels = flattenBreakdown(labels)
+      metrics.getOrCreateCounter('span.self_time.count', labels).inc()
+      metrics.getOrCreateCounter('span.self_time.sum.us', labels).inc(value)
+    })
+  }
+}
+
+function transactionBreakdownDetails ({ name, type } = {}) {
+  return {
+    name,
+    type
+  }
+}
+
+function spanBreakdownDetails (span) {
+  if (span instanceof Transaction) {
+    return {
+      type: 'app'
+    }
+  }
+
+  const [ type, subtype ] = span.type.split('.')
+  return {
+    type,
+    subtype
+  }
+}
+
+function captureBreakdown (transaction, labels, time) {
+  const build = () => ({ labels, value: 0 })
+  const counter = transaction._breakdownTimings.ensure(labels, build)
+  counter.value += time
```

Comment:
Seems like you are not capturing the count per span type. For example when there are two `db.mysql` spans, each taking 1µs, the metricset should have `span.self_time.count=2` and `span.self_time.sum.us`.

But maybe I'm just not understanding how the ObjectIdentityMap works. If that contains two different counters in that example for the same (equal) key (with different object instances), it should work.

Reply:
Ah, no. It's only counting the time, not the number occurrences, within the request. I'll fix that. 😅

Reply:
Fixed